upload artifact per build for easier review

jbkoh opened the pull request.

gtfierro commented on .github/workflows/builds.yml:
```diff
@@ -39,6 +36,14 @@ jobs:
+        name: ontology
+        path: |
+          Brick.ttl
+          shacl/BrickShape.ttl
```
we don't need to include the `BrickShape.ttl` file ; it should be included in the `Brick.ttl` build.

gtfierro reviewed (commented):
Thanks for the PR! This is a nice idea. Do you know where the artifacts get uploaded?

gtfierro commented on .github/workflows/builds.yml:
```diff
@@ -6,9 +6,6 @@ on:
         - master
         - v1.*
   pull_request:
-    branches:
-        - master
-        - v1.*
```
Does leaving this empty mean that it triggers on *all* PRs?

jbkoh replied:
Yeah, that is intended. Every PR deserves a build and automated validation. Are there reasons not to do so?

gtfierro replied:
I agree with you that we want a build on all PRs... my question is whether or not this was the correct GH action syntax to accomplish that

jbkoh replied:
Sorry, misunderstood. Yes, it triggers on all PRs whose activit type is one of opened, synchronize, or reopened.

> Note: More than one activity type triggers this event. For information about each activity type, see "[Webhook events and payloads](https://docs.github.com/en/github-ae@latest/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#pull_request)." By default, a workflow only runs when a pull_request event's activity type is opened, synchronize, or reopened. You can specify different activity types using the types keyword. For more information, see "[Workflow syntax for GitHub Actions](https://docs.github.com/en/github-ae@latest/articles/workflow-syntax-for-github-actions#onevent_nametypes)."

jbkoh commented:
> Thanks for the PR! This is a nice idea. Do you know where the artifacts get uploaded?

You can find artifacts at the Summary of each Action. E.g.,
![image](https://user-images.githubusercontent.com/1572627/162368612-cd5dbbbc-42f7-454b-ab5d-704395490bfc.png)

gtfierro commented on .github/workflows/builds.yml:
```diff
@@ -39,6 +36,13 @@ jobs:
+      if: ${{ matrix.python-version }} == "3.7.9"
+      uses: actions/upload-artifact@v3
+      with:
+        name: ontology
```
Can we change the name of the artifact to "Brick-files" or something more descriptive and Brick-specific than "ontology"? "Brick-ontology" would be fine as well

jbkoh commented:
@gtfierro I just updated the name. You were right that artifacts are always zipped at the UI: https://github.com/actions/upload-artifact#zipped-artifact-downloads, though those files are not zipped in their system. I updated the filename. Please let me know if it's ready to be merged.

gtfierro commented:
Looks great! I"ll merge it in

gtfierro commented:
I'm not questioning whether this is a good idea; I agree with you this is what we want to do. My question is whether or not this is the correct syntax for accomplishing having a build on every pull request. I would think there would be some sort of glob pattern, rather than just leaving the trigger blank?


________________________________
From: Jason B. Koh ***@***.***>
Sent: Thursday, April 7, 2022 11:05:38 PM
To: BrickSchema/Brick ***@***.***>
Cc: Gabriel Fierro ***@***.***>; Review requested ***@***.***>
Subject: [External] Re: [BrickSchema/Brick] upload artifact per build for easier review (PR #388)


@jbkoh commented on this pull request.

________________________________

In .github/workflows/builds.yml<https://nam04.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FBrickSchema%2FBrick%2Fpull%2F388%23discussion_r845739999&data=04%7C01%7Cgtfierro%40mines.edu%7Cbda54bba843144722ca408da191d6c35%7C997209e009b346239a4d76afa44a675c%7C0%7C0%7C637849911419915170%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=IJ27nLfUsrr0Fi1USmWVDKVgocbMLDXYJE885wipU9I%3D&reserved=0>:

> -    branches:
-        - master
-        - v1.*


Yeah, that is intended. Every PR deserves a build and automated validation. Are there reasons not to do so?

—
Reply to this email directly, view it on GitHub<https://nam04.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FBrickSchema%2FBrick%2Fpull%2F388%23discussion_r845739999&data=04%7C01%7Cgtfierro%40mines.edu%7Cbda54bba843144722ca408da191d6c35%7C997209e009b346239a4d76afa44a675c%7C0%7C0%7C637849911419915170%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=IJ27nLfUsrr0Fi1USmWVDKVgocbMLDXYJE885wipU9I%3D&reserved=0>, or unsubscribe<https://nam04.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAAANRCS3V6MDAHSFUGYXW23VD65CFANCNFSM5S2UD4CA&data=04%7C01%7Cgtfierro%40mines.edu%7Cbda54bba843144722ca408da191d6c35%7C997209e009b346239a4d76afa44a675c%7C0%7C0%7C637849911419915170%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=yAnynGNl3P1D7xX0tZrxMi%2F8%2FuzIkRr5L%2FDz%2Bm8HJG8%3D&reserved=0>.
You are receiving this because your review was requested.Message ID: ***@***.***>

jbkoh commented:
I believe the syntax is correct, but I can take a look later; I'm traveling now. Have you seen unexpected behaviors?
